File: Fib.java
package start;

public class Fib {
	public static void main(String[] args) {
		p(0);
		int a = 1;
		p(a);	
		p(a);
		int b = a + a;
		p(b);
		while(a < 300) {
			a = a + b;
			p(a);
			b = a + b;
			p(b);
		}
	}
	public static void p(int outp) {
		System.out.println(outp);
	}
}

File: fib.ml
let rec fib =
  let cache = Hashtbl.create 64 in
  fun n ->
    if n < 2 then n
    else
      try Hashtbl.find cache n
      with Not_found ->
        let result = fib (n - 1) + fib (n - 2) in
        Hashtbl.add cache n result ; result

let () =
  let rec loop i =
    let a = fib i in
    Format.printf "%d@.%d@." a (fib (i + 1)) ;
    if a < 1000000 then loop (i + 2)
  in
  loop 0
